channels/routing.py

```diff
@@ -135,9 +108,10 @@ async def __call__(self, scope, receive, send):
         # Run through the routes we have until one matches
         for route in self.routes:
             try:
-                match = route_pattern_match(route, path)
+                match = route.pattern.match(path)
                 if match:
                     new_path, args, kwargs = match
+                    kwargs.update(route.default_args)
```

Comment:
What's this for? 🤔

Reply:
Ok, so the example is this: 

```
            path("defaultkwargs/", defaultkwarg_app, kwargs={"default": 42}),
```

So we're adding the `kwargs={"default": 42}` — added a comment in https://github.com/django/channels/pull/1894/commits/0276b612c695a040f8ee4ffddbf0240e0ff9421c

Reply:
Yes, exactly. The consumer should receive the merged arguments from all patterns *and* from the data passed to `path` (or `re_path`)